Add TableScanBuilder

majetideepak opened the pull request:
Remove various tableScan APIs.
Add hiveTableScan API to build a Hive Connector TableScanNode. This is sufficient for most use-cases.
Add tpchTableScan API to build a TPC-H Connector TableScanNode.
Add TableScanBuilder API to build a custom TableScanNode.

Resolves: https://github.com/facebookincubator/velox/issues/7391

majetideepak commented:
@mbasmanova, this is ready for review.

mbasmanova commented on velox/exec/tests/utils/PlanBuilder.h:
```diff
@@ -132,45 +107,139 @@ class PlanBuilder {
+    }
+
+    /// @param outputType List of column names and types to read from the table.
+    TableScanBuilder& outputType(const RowTypePtr outputType) {
```
drop const + std::move

mbasmanova commented on velox/exec/tests/utils/PlanBuilder.h:
```diff
@@ -132,45 +107,139 @@ class PlanBuilder {
+    /// column < value, column >= value, column > value, column = value, column
+    /// IN (v1, v2,.. vN), column < v1 OR column >= v2.
+    TableScanBuilder& subfieldFilters(
+        const std::vector<std::string>& subfieldFilters) {
```
remove const & + std::move

mbasmanova commented on velox/exec/tests/utils/PlanBuilder.h:
```diff
@@ -132,45 +107,139 @@ class PlanBuilder {
+    /// @param remainingFilter SQL expression for the additional conjunct. May
+    /// include multiple columns and SQL functions. The remainingFilter is
+    /// AND'ed with all the subfieldFilters.
+    TableScanBuilder& remainingFilter(const std::string& remainingFilter) {
```
ditto

mbasmanova commented on velox/exec/tests/utils/PlanBuilder.h:
```diff
@@ -132,45 +107,139 @@ class PlanBuilder {
+      return *this;
+    }
+
+    /// @param tableHandle
```
Redundant comment. Let's remove.

mbasmanova commented on velox/exec/tests/utils/PlanBuilder.h:
```diff
@@ -132,45 +107,139 @@ class PlanBuilder {
+      return *this;
+    }
+
+    /// @param assignments
```
broken formatting?

mbasmanova commented on velox/exec/tests/utils/PlanBuilder.h:
```diff
@@ -132,45 +107,139 @@ class PlanBuilder {
+    }
+
+    /// Build the plan node TableScanNode.
+    core::PlanNodePtr build();
```
Should this method be private?

mbasmanova commented on velox/exec/tests/utils/PlanBuilder.h:
```diff
@@ -132,45 +107,139 @@ class PlanBuilder {
+
+   private:
+    PlanBuilder& planBuilder_;
+    core::PlanNodeId id_;
```
Is this needed? Can it be a local variable in endTableScan?

mbasmanova reviewed (commented):
@majetideepak Thank you for working on this. This change is very large. Is there a way to break it up into smaller pieces?

majetideepak commented:
@mbasmanova I opened a smaller PR here https://github.com/facebookincubator/velox/pull/7606 and addressed your comments there.
I will add other changes in subsequent PRs.

majetideepak commented:
This work has been merged as two other PRs
